File: CNEASY.cpp
#include <bits/stdc++.h>

using namespace std;

int main()
{
	int tc;
	cin >> tc;
	while(tc--)
	{
		int n;
		cin >> n;
		string s;
		double a[n],x;
		for (int i = 0; i < n; ++i)
			cin >> s >> a[i];
		sort(a,a+n);
		double la = a[n-1] - a[0];
		for (int i = 1; i < n; ++i)
		{
			x = 360.0 - a[i] + a[i-1];
			if(x < la)
				la = x;
		}	
		la *= 12.0;
		int ans = ceil(la);
		cout << ans << "\n";
	}
	return 0;
}

File: CNEASY.py
from math import ceil

tc = input()
while tc > 0:
	n = input()
	val = []
	for x in xrange(0,n):
		val.append(float(raw_input().split()[1]))
	val.sort()
	la = float(val[n-1] - val[0])
	for x in xrange(1,n):
		tmp = 360.0 - val[x] + val[x-1]
		la = min(la,tmp)
	la *= 12.0
	print "%d" % int(ceil(la))
	tc -= 1